### PR TITLE
fix(core): prevent app permissions from being lost during an app upse…

### PR DESCRIPTION
**Pull request by danielpeach**
…rt operation

### CODE REVIEW

File: orca-applications/src/main/groovy/com/netflix/spinnaker/orca/applications/tasks/UpsertApplicationTask.groovy [hunk 1]
```diff
@@ -44,10 +45,15 @@ class UpsertApplicationTask extends AbstractFront50Task {
     } else {
       log.info("Creating application (name: ${application.name})")
       front50Service.create(application)
+      if (application?.permission?.permissions == null) {
```
Comment (lwander) [inline orca-applications/src/main/groovy/com/netflix/spinnaker/orca/applications/tasks/UpsertApplicationTask.groovy]: You can probably groovify this with `if (!application?.permission?.permissions)`

File: orca-applications/src/main/groovy/com/netflix/spinnaker/orca/applications/tasks/UpsertApplicationTask.groovy [hunk 2]
```diff
@@ -44,10 +45,15 @@ class UpsertApplicationTask extends AbstractFront50Task {
...
+        application.setPermissions(Permissions.EMPTY)
+      }
     }
 
     try {
-      front50Service.updatePermission(application.name, application.permission)
+      if (application?.permission?.permissions != null) {
```
Comment (lwander) [inline orca-applications/src/main/groovy/com/netflix/spinnaker/orca/applications/tasks/UpsertApplicationTask.groovy]: Similar here to above

File: orca-applications/src/main/groovy/com/netflix/spinnaker/orca/applications/tasks/UpsertApplicationTask.groovy (hunk 1 above)
Comment (danielpeach) [reply orca-applications/src/main/groovy/com/netflix/spinnaker/orca/applications/tasks/UpsertApplicationTask.groovy]: I want this condition to be true for an empty map - `null` should mean no change to the permissions, whereas `[:]` should mean an empty set of permissions.

Comment (lwander) [reply orca-applications/src/main/groovy/com/netflix/spinnaker/orca/applications/tasks/UpsertApplicationTask.groovy]: It'll also be true if `application` or `permission` are `null` when written like this -- is that intended?

Comment (danielpeach) [reply orca-applications/src/main/groovy/com/netflix/spinnaker/orca/applications/tasks/UpsertApplicationTask.groovy]: Yes for `permission`, no for `application`, so I'll take out the null check for application.

**Comment by ttomsu** [comment]
LGTM

**Pull request merged**